### PR TITLE
Fix GCC 9 shadowing compiler warnings in Core (2nd try)

### PR DESCRIPTION
**Pull request by wddgit**
#### PR description:

Fix compiler warnings in Core that appear when GCC 9 shadowing
warnings are turned on.

These are fixed in several ways:

Some names are changed to be more descriptive (this is my second try at
this, in the original PR I often disambiguated names by adding numerical
suffixes but there is none of that in this PR)

Sometimes I added curly braces in appropriate places to put code
in separate blocks where that made sense.

In a few places in the tests, I broke large functions into smaller independent
functions.

Sometimes after carefully looking at the code I realized things could be
completely deleted or refactored in some way.

A little cleanup is mixed up with this. If I saw something that looked bad
in the vicinity of the shadowing fixes.

#### PR validation:

This relies on existing tests. Nothing should change in the externally
visible behavior of the code.

### CODE REVIEW

**Comment by wddgit** [comment]
please test

**Comment by wddgit** [comment]
please test

Fix more warnings. (I forgot to compile again at the end with the shadow warnings enabled in my GCC 9 area to verify I fixed them all. When I did I found some more...)

File: IOPool/Input/src/RootEmbeddedFileSequence.cc [hunk 1]
```diff
@@ -302,7 +302,7 @@ namespace edm {
       found = rootFile()->readCurrentEvent(cache);
     }
     if (!found) {
-      bool found = rootFile()->setEntryAtItem(id.run(), id.luminosityBlock(), 0);
+      found = rootFile()->setEntryAtItem(id.run(), id.luminosityBlock(), 0);
```
Comment (Dr15Jones) [inline IOPool/Input/src/RootEmbeddedFileSequence.cc]: OK

**Comment by Dr15Jones** [comment]
please test

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp [hunk 2]
```diff
@@ -251,8 +240,8 @@ namespace {
 
 void testServiceRegistry::threadTest() {
   UniqueRegistry::isUnique_ = false;
-  void* value = &(edm::ServiceRegistry::instance());
-  UniqueRegistry unique(value);
+  void* serviceRegistry = &(edm::ServiceRegistry::instance());
+  UniqueRegistry unique(serviceRegistry);
```
Comment (Dr15Jones) [inline FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: OK

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp [hunk 3]
```diff
@@ -73,75 +73,64 @@ namespace {
...
-          edm::ServiceRegistry::createContaining(std::move(dummyPtr), token, edm::serviceregistry::kOverlapIsError));
-
-      edm::ServiceRegistry::Operate operate(token2);
-      edm::Service<testserviceregistry::DummyService> dummy;
-      CPPUNIT_ASSERT(dummy);
-      CPPUNIT_ASSERT(dummy.isAvailable());
-      CPPUNIT_ASSERT(dummy->value() == 2);
-    }
-  }
+  {{auto dummyPtr = std::make_unique<DummyService>();
```
Comment (Dr15Jones) [inline FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: This spacing looks weird. Did clang format do this?

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp [hunk 4]
```diff
@@ -73,75 +73,64 @@ namespace {
...
+  dummyPtr->value_ = 2;
+  edm::ServiceToken token(edm::ServiceRegistry::createContaining(std::move(dummyPtr)));
```
Comment (Dr15Jones) [inline FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: This change is incorrect. the `token` is meant to have a lifetime that overlaps `token2` which is in the next scope.

File: FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc [hunk 5]
```diff
@@ -39,33 +36,6 @@ class testintersectingiovrecordintervalfinder : public CppUnit::TestFixture {
...
-        } else {
-          iInterval = edm::ValidityInterval();
-        }
-      }
-    }
-
-  private:
-    edm::ValidityInterval interval_;
-  };
-}  // namespace
```
Comment (Dr15Jones) [inline FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc]: Not needed so OK to remove.

File: FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc [hunk 6]
```diff
@@ -77,116 +47,96 @@ void testintersectingiovrecordintervalfinder::constructorTest() {
...
-    const edm::EventID eID_5(1, 1, 5);
-    const edm::IOVSyncValue sync_5(eID_5);
-    const edm::ValidityInterval unknownedEndInterval(sync_5, edm::IOVSyncValue::invalidIOVSyncValue());
+    const edm::ValidityInterval unknownedEndInterval(edm::IOVSyncValue(edm::EventID(1, 1, 5)),
+                                                     edm::IOVSyncValue::invalidIOVSyncValue());
     dummyFinder->setInterval(unknownedEndInterval);
 
     CPPUNIT_ASSERT(unknownedEndInterval ==
-                   intFinder.findIntervalFor(dummyRecordKey, edm::IOVSyncValue(edm::EventID(1, 1, 5))));
+                   intersectingFinder.findIntervalFor(dummyRecordKey, edm::IOVSyncValue(edm::EventID(1, 1, 5))));
```
Comment (Dr15Jones) [inline FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc]: OK, do not need another scope

File: FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc [hunk 7]
```diff
@@ -77,116 +47,96 @@ void testintersectingiovrecordintervalfinder::constructorTest() {
...
   {
     finders.clear();
-    const edm::EventID eID_1(1, 1, 1);
-    const edm::IOVSyncValue sync_1(eID_1);
-    const edm::EventID eID_3(1, 1, 3);
-    const edm::IOVSyncValue sync_3(eID_3);
-    const edm::EventID eID_4(1, 1, 4);
-    const edm::IOVSyncValue sync_4(eID_4);
-    const edm::EventID eID_5(1, 1, 5);
-    const edm::IOVSyncValue sync_5(eID_5);
```
Comment (Dr15Jones) [inline FWCore/Framework/test/intersectingiovrecordintervalfinder_t.cppunit.cc]: OK, 5 not used

File: FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc [hunk 8]
```diff
@@ -168,8 +168,8 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
 
     REQUIRE(mergeableRunProductProcesses.size() == 2);
 
-    std::vector<std::string> expected{"AAPROD", "APROD"};
-    REQUIRE(mergeableRunProductProcesses.processesWithMergeableRunProducts() == expected);
+    std::vector<std::string> expectedProcesses{"AAPROD", "APROD"};
```
Comment (Dr15Jones) [inline FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc]: Try moving these two lines into their own scope to limit lifetime of `expectedProcesses`

File: FWCore/ParameterSet/test/parameterSetDescription_t.cc [hunk 9]
```diff
@@ -1626,7 +1654,7 @@ int main(int, char**) try {
     nestLevel1.add<int>("intLevel1a", 1);
     nestLevel1.add<edm::ParameterSetDescription>("nestLevel1b", nestLevel2);
 
-    testDescriptions[i].addVPSetUntracked("nestLevel0", nestLevel1);
+    testDescriptions[ii].addVPSetUntracked("nestLevel0", nestLevel1);
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/test/parameterSetDescription_t.cc]: OK

File: FWCore/ParameterSet/test/parameterSetDescription_t.cc [hunk 10]
```diff
@@ -924,172 +924,200 @@ namespace testParameterSetDescription {
...
-
-    edm::ParameterSet pset1;
+    {
+      std::unique_ptr<edm::ParameterDescriptionNode> node(
+          std::make_unique<edm::AllowedLabelsDescription<int>>("allowedLabels", true));
 
-    edm::ParameterSet pset2;
-    std::vector<std::string> labels;
-    pset2.addParameter<std::vector<std::string>>("allowedLabels", labels);
+      edm::ParameterSet emptyPset;
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/test/parameterSetDescription_t.cc]: Can this be made const?

File: FWCore/ParameterSet/test/parameterSetDescription_t.cc [hunk 11]
```diff
@@ -924,172 +924,200 @@ namespace testParameterSetDescription {
...
+          assert(0);
+        } catch (edm::Exception const&) { /* There should be an exception */
+        }
+      }
+      {
+        edm::ParameterSetDescription psetDesc;
+        psetDesc.labelsFrom<std::vector<edm::ParameterSet>>(std::string("allowedLabelsC"), nestedPSetDesc);
+        psetDesc.validate(pset);
       }
-      psetDesc20.validate(psetC);
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/test/parameterSetDescription_t.cc]: NOTE: At some future point, this really should be moved to either use an actual unit test framework.

**Comment by Dr15Jones** [comment]
I am done with the initial review.

File: FWCore/Concurrency/test/limitedtaskqueue_t.cppunit.cpp [hunk 12]
```diff
@@ -253,27 +253,26 @@ void LimitedTaskQueue_test::stressTest() {
...
+            auto nrt = nRunningTasks++;
+            if (nrt >= kMax) {
+              std::cout << "ERROR " << nRunningTasks << " >= " << kMax << std::endl;
             }
+            CPPUNIT_ASSERT(nrt < kMax);
+            ++count;
+            --nRunningTasks;
           });
+        }
+      });
```
[the remark this answers is not in the document]
Comment (wddgit) [reply FWCore/Concurrency/test/limitedtaskqueue_t.cppunit.cpp]: My only change was "guard" to "guardAgain" (line 265 in new version).

All the rest is spacing. I have no idea why "scram code-format" put an
extra line break in line 256 that it didn't put there before. And that
caused a bunch of spacing changes after that.

**Comment by Dr15Jones** [comment]
@slava77 @perrotta any ideas why workflow 11634 is showing differences in Conversions and PFBlocks?

File: FWCore/ParameterSet/interface/PluginDescription.h [hunk 13]
```diff
@@ -153,18 +153,16 @@ namespace edm {
...
-      std::stringstream ss;
-      ss << dfh.section() << "." << dfh.counter();
-      std::string newSection = ss.str();
-
-      printSpaces(os, indentation);
-      os << "Section " << newSection << " " << Factory::get()->category() << " Plugins description:\n";
-      if (!dfh.brief())
-        os << "\n";
-
-      DocFormatHelper new_dfh(dfh);
```
[the remark this answers is not in the document]
Comment (wddgit) [reply FWCore/ParameterSet/interface/PluginDescription.h]: This "new_dfh" object serves no purpose. It is created and never used.
I imagine it was originally a harmless cut and paste error.

**Comment by perrotta** [comment]
@Dr15Jones I have noticed the same in other unrelated PR's. The number of converted photons in that test jump from 29 and 30.
The first time I noticed it was in IBs of Sept 20, by the way the same baseline IB used here.
In all other cases, when I rerun the tests (after Sept 22) the differences did not show up any more.
No idea where it comes from: if there is no obvious culprit merged right before Sept 20 to investigate, maybe some check with Valgrind should be done.

File: FWCore/Framework/test/DummyFinder.h [hunk 14]
```diff
@@ -18,19 +18,19 @@
...
-
 class DummyFinder : public edm::EventSetupRecordIntervalFinder {
 public:
   DummyFinder() : edm::EventSetupRecordIntervalFinder(), interval_() { this->findingRecord<DummyRecord>(); }
 
-  void setInterval(const edm::ValidityInterval& iInterval) { interval_ = iInterval; }
+  void setInterval(const edm::ValidityInterval& iInterval) {
+    interval_ = iInterval;
+    const edm::eventsetup::EventSetupRecordKey dummyRecordKey = DummyRecord::keyForClass();
+    resetInterval(dummyRecordKey);
```
[the remark this answers is not in the document]
Comment (wddgit) [reply FWCore/Framework/test/DummyFinder.h]: I agree it is not a shadow fix. This fixes a potential bug, although DummyFinder
 is not actually used in a way that hits the bug. In the unit
test, setInterval changes the interval DummyFinder uses to calculate what
to return from setIntervalFor, but DummyFinder also has cached a value
based on the old interval. Without this fix the cached value could
potentially get used in setIntervalFor even though it isn't valid anymore.

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp (hunk 3 above)
Comment (wddgit) [reply FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: clang format did this spacing. I don't like it either. Just now, I tried
backing it out and redoing the indentation as I like it, but that
didn't help. At least in my working area, "scram b code-format"
put this weird formatting back in. I do not see any problem in my
code causing it. I would like to see every open curly brace on its
own line with increasing  indentation inside each one...

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp (hunk 4 above)
Comment (wddgit) [reply FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: In the old version, "token" on line 79 is shadowing the "token" on
line 97.  I can change it to keep the token on line 79 alive if that
serves some test purpose internally. It wasn't obvious to me that
it was.

**Comment by Dr15Jones** [comment]
@perrotta Thanks!

File: FWCore/Framework/src/Principal.cc [hunk 15]
```diff
@@ -445,10 +445,10 @@ namespace edm {
       // The current process might be needed but not be in the process
       // history if all the products produced in the current process are
       // transient.
-      auto nameIter =
+      auto nameIterCurrentProcess =
```
[the remark this answers is not in the document]
Comment (wddgit) [reply FWCore/Framework/src/Principal.cc]: Done

File: FWCore/ParameterSet/src/ParameterSetConverter.cc [hunk 16]
```diff
@@ -157,9 +157,9 @@ namespace edm {
       }
     }
     if (!doItAgain && !parameterSets_.empty()) {
-      for (StringWithIDList::iterator i = parameterSets_.begin(), iEnd = parameterSets_.end(); i != iEnd; ++i) {
+      for (StringWithIDList::iterator k = parameterSets_.begin(), kEnd = parameterSets_.end(); k != kEnd; ++k) {
```
[the remark this answers is not in the document]
Comment (wddgit) [reply FWCore/ParameterSet/src/ParameterSetConverter.cc]: Done

File: FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp (hunk 4 above)
Comment (wddgit) [reply FWCore/ServiceRegistry/test/serviceregistry_t.cppunit.cpp]: Done. Fixed this so the token is still alive. Note this also had the side effect of fixing the weird formatting discussed above...

File: FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc (hunk 8 above)
Comment (wddgit) [reply FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc]: Done

File: FWCore/ParameterSet/test/parameterSetDescription_t.cc (hunk 10 above)
Comment (wddgit) [reply FWCore/ParameterSet/test/parameterSetDescription_t.cc]: Done

**Comment by wddgit** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**